homeassistant/components/eq3btsmart/entity.py

```diff
@@ -1,19 +1,17 @@
 """Base class for all eQ-3 entities."""
 
-from eq3btsmart.thermostat import Thermostat
-
 from homeassistant.helpers.entity import Entity
 
-from .models import Eq3Config
+from . import Eq3ConfigEntry
 
 
 class Eq3Entity(Entity):
     """Base class for all eQ-3 entities."""
 
     _attr_has_entity_name = True
 
-    def __init__(self, eq3_config: Eq3Config, thermostat: Thermostat) -> None:
+    def __init__(self, entry: Eq3ConfigEntry) -> None:
```

Comment:
I am personally not really a fan of putting in the full config entry object in the entities as it always feels like a big object and I like to keep the entities simple and just put in as little as possible. (but this is a opinionated comment, feel free to ignore)

Reply:
This was done because of this requested change https://github.com/home-assistant/core/pull/130263#discussion_r1835739261

I personally prefer passing the config entry now that it's using runtime data.